src/backend/distributed/worker/worker_data_fetch_protocol.c

```diff
@@ -791,9 +795,9 @@ FetchForeignTable(const char *nodeName, uint32 nodePort, StringInfo tableName)
 
 	/* fetch foreign file to this node in an idempotent manner */
 	localFilePath = makeStringInfo();
-	appendStringInfo(localFilePath, FOREIGN_CACHED_FILE_PATH, tableName->data);
+	appendStringInfo(localFilePath, FOREIGN_CACHED_FILE_PATH, tableName);
```

Comment:
Shouldn't this have the schema included in the file name somehow? What if there are two foreign tables with the same name in different schemas? Wouldn't they conflict and end up using the same file?

Maybe I'm missing something, but I think this is a place that the schema needs to appear, too. It doesn't look possible to make a directory, so maybe the schema can be included in the file name somehow.